CMakeLists.txt

```diff
@@ -238,7 +238,15 @@ endif()
 # =============================================================================
 if(CORENRN_HAVE_NVHPC_COMPILER)
   # PGI with llvm code generation doesn't have necessary assembly intrinsic headers
-  add_definitions(-DEIGEN_DONT_VECTORIZE=1)
+  list(APPEND CORENRN_COMPILE_DEFS EIGEN_DONT_VECTORIZE=1)
+  if(NOT CORENRN_ENABLE_GPU AND ${CMAKE_CXX_COMPILER_VERSION} VERSION_GREATER_EQUAL 21.11)
+    # Random123 does not play nicely with NVHPC 21.11+'s detection of ABM features if it detects the
+    # compiler to be PGI or NVHPC, see: https://github.com/BlueBrain/CoreNeuron/issues/724 and
+    # https://github.com/DEShawResearch/random123/issues/6. In fact in GPU builds Random123
+    # (mis)detects nvc++ as nvcc because we pass the -cuda option and we therefore avoid the
+    # problem. If GPU support is disabled, we define R123_USE_INTRIN_H=0 to avoid the problem.
+    list(APPEND CORENRN_COMPILE_DEFS R123_USE_INTRIN_H=0)
+  endif()
 endif()
```

Comment:
@olupton : FYI - (accidentally) I was trying to use cpu wheel with nvhpc compiler and end-up with this error. 

Wonder if this logic needs to be pushed at later stage (in `nrnivmodl_core_makefile` or similar)